spec/support/http_handling_shared.rb

```diff
@@ -148,6 +148,18 @@
         expect(sockets_used.uniq.length).to eq(1)
       end
 
+      context "on a mixed state" do
+        it "re-opens the connection" do
+          first_socket = client.get("#{server.endpoint}/socket/1").body.to_s
+
+          client.instance_variable_set(:@state, :dirty)
```

Comment:
I wish there were a way to test it without patching internal state. Like enforce timeout... :D